src/component-library/Table/Table.tsx

```diff
@@ -1,39 +1,27 @@
-import { ReactNode } from 'react';
+import { Cell, Column, Row, TableBody, TableHeader } from '@react-stately/table';
+import { forwardRef, ReactNode } from 'react';
 
-import { StyledTable, TableCell, TableHeader, TableRow, TableWrapper } from './Table.style';
+import { BaseTable, BaseTableProps } from './BaseTable';
 
-type Cell = ReactNode;
-type Row = Cell[];
+type Props = {
+  columns: Array<{ name: ReactNode; uid: string }>;
+  rows: Array<Record<string, any>>;
+};
 
-interface TableProps {
-  columnLabels: string[];
-  rows: Row[];
-}
+type InheritAttrs = Omit<BaseTableProps, keyof Props | 'children'>;
 
-const Table = ({ columnLabels, rows }: TableProps): JSX.Element => {
-  return (
-    <TableWrapper>
-      <StyledTable>
-        <thead>
-          <TableRow>
-            {columnLabels.map((columnLabel) => (
-              <TableHeader key={columnLabel}>{columnLabel}</TableHeader>
-            ))}
-          </TableRow>
-        </thead>
-        <tbody>
-          {rows.map((row, rowIndex: number) => (
-            <TableRow key={`row-${rowIndex}`}>
-              {row.map((cell, cellIndex) => (
-                <TableCell key={`row-${rowIndex}-cell${cellIndex}`}>{cell}</TableCell>
-              ))}
-            </TableRow>
-          ))}
-        </tbody>
-      </StyledTable>
-    </TableWrapper>
-  );
-};
+type TableProps = Props & InheritAttrs;
+
+const Table = forwardRef<HTMLTableElement, TableProps>(
+  ({ columns, rows, ...props }, ref): JSX.Element => (
+    <BaseTable ref={ref} {...props}>
+      <TableHeader columns={columns}>{(column) => <Column key={column.uid}>{column.name}</Column>}</TableHeader>
+      <TableBody items={rows}>{(item: any) => <Row>{(columnKey) => <Cell>{item[columnKey]}</Cell>}</Row>}</TableBody>
```

Comment:
These are not UI Component full solely components that hold logic necessary for their hooks.

https://github.com/adobe/react-spectrum/blob/main/packages/%40react-stately/table/src/Cell.ts